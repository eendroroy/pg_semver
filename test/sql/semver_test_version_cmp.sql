LOAD 'pg_semver';

SELECT PG_SEMVER_CMP('0.0.0', '0.0.1');
SELECT PG_SEMVER_CMP('0.0.1', '0.0.1');
SELECT PG_SEMVER_CMP('0.0.2', '0.0.1');

SELECT PG_SEMVER_CMP('1.0.0-alpha.1', '1.0.0-alpha.0');
SELECT PG_SEMVER_CMP('1.0.0-alpha.1', '1.0.0-alpha.2');
SELECT PG_SEMVER_CMP('1.0.0-alpha.1', '1.0.0-beta.2');
SELECT PG_SEMVER_CMP('1.0.0-rc.1', '1.0.0');

SELECT PG_SEMVER_CMP('1.0.0-rc.1*', '1.0.0');
SELECT PG_SEMVER_CMP('1.0.0-rc.@1', '1.0.0');
SELECT PG_SEMVER_CMP('1.0.0-rc.#1', '1.0.0');
