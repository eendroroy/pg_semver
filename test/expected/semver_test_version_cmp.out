LOAD 'pg_semver';
SELECT PG_SEMVER_CMP('0.0.0', '0.0.1');
 pg_semver_cmp 
---------------
            -1
(1 row)

SELECT PG_SEMVER_CMP('0.0.1', '0.0.1');
 pg_semver_cmp 
---------------
             0
(1 row)

SELECT PG_SEMVER_CMP('0.0.2', '0.0.1');
 pg_semver_cmp 
---------------
             1
(1 row)

SELECT PG_SEMVER_CMP('1.0.0-alpha.1', '1.0.0-alpha.0');
 pg_semver_cmp 
---------------
             1
(1 row)

SELECT PG_SEMVER_CMP('1.0.0-alpha.1', '1.0.0-alpha.2');
 pg_semver_cmp 
---------------
            -1
(1 row)

SELECT PG_SEMVER_CMP('1.0.0-alpha.1', '1.0.0-beta.2');
 pg_semver_cmp 
---------------
            -1
(1 row)

SELECT PG_SEMVER_CMP('1.0.0-rc.1', '1.0.0');
 pg_semver_cmp 
---------------
            -1
(1 row)

SELECT PG_SEMVER_CMP('1.0.0-rc.1*', '1.0.0');
ERROR:  Invalid Semver '1.0.0-rc.1*'.
LINE 1: SELECT PG_SEMVER_CMP('1.0.0-rc.1*', '1.0.0');
                             ^
DETAIL:  Semver '1.0.0-rc.1*' contains invalid character.
HINT:  Did you mean '1.0.0-rc.1'?
SELECT PG_SEMVER_CMP('1.0.0-rc.@1', '1.0.0');
ERROR:  Invalid Semver '1.0.0-rc.@1'.
LINE 1: SELECT PG_SEMVER_CMP('1.0.0-rc.@1', '1.0.0');
                             ^
DETAIL:  Semver '1.0.0-rc.@1' contains invalid character.
HINT:  Did you mean '1.0.0-rc.1'?
SELECT PG_SEMVER_CMP('1.0.0-rc.#1', '1.0.0');
ERROR:  Invalid Semver '1.0.0-rc.#1'.
LINE 1: SELECT PG_SEMVER_CMP('1.0.0-rc.#1', '1.0.0');
                             ^
DETAIL:  Semver '1.0.0-rc.#1' contains invalid character.
HINT:  Did you mean '1.0.0-rc.1'?
